assets/js/app.js

```diff
@@ -0,0 +1,277 @@
+                            document.getElementById("choice3").innerHTML = fiveCoinsInStorage[2].slice(6)
+                            document.getElementById("choice4").innerHTML = fiveCoinsInStorage[3].slice(6)
+                            document.getElementById("choice5").innerHTML = fiveCoinsInStorage[4].slice(6)
+
+                            //add an event listener for the erasing options
+                            const fiveChoices = document.querySelectorAll("#choice1, #choice2, #choice3, #choice4, #choice5")
+                            fiveChoices.forEach(function (choice) {
+                                choice.addEventListener(`click`, function () {
+                                    const indexToRemove = fiveCoinsInStorage.findIndex(coin => coin === `switch${choice.innerHTML}`)
+                                    fiveCoinsInStorage.splice(indexToRemove, 1) //remove to chosen coin to remove
+                                    fiveCoinsInStorage.push(input.id) //add the new input to the array
+                                    localStorage.setItem(`5coinsInStorage`, JSON.stringify(fiveCoinsInStorage)) //save
+                                    document.getElementById(`switch${choice.innerHTML}`).checked = false //update the DOM
+                                    myModal.hide() //hide again the modal
+                                })
+
+                                const choicesClosed = document.getElementById("choicesClosed") //if the user chose to close the replacing modal
+                                choicesClosed.addEventListener(`click`, () => {
+                                    input.checked = false //making the input false again
+                                })
+                            })
+                            //finally, show the modal, once all the settings are finished
+                            myModal.show()
+                        }
+                        else fiveCoinsInStorage.push(input.id) //if there isn`t 5 coins in storage
+                    } else { //in case the change in the status of the button was to delete coin from selection
+                        const indexToRemove = fiveCoinsInStorage.findIndex(coin => coin.id === input.id)
+                        fiveCoinsInStorage.splice(indexToRemove, 1)
+                    }
+                    localStorage.setItem(`5coinsInStorage`, JSON.stringify(fiveCoinsInStorage)) //save
+                })
+            })
+        }
+        allTheFunctionsForCoinPage()
+
+        //the search option
+        const search = document.getElementById("searchButton")
+        search.addEventListener(`click`, async function () {
+            const searchInput = document.getElementById("searchInput").value
+
+            const coinsCache = JSON.parse(localStorage.getItem(`coinsCache`))
+            const index = coinsCache.findIndex(coin => coin.id === searchInput)
```

Comment:
this is a very restrictive search. should have be more flexible with `includes` instead of `===`